scanner/line_scanner.coffee

```diff
@@ -159,13 +159,17 @@ class LineScanner
   extractedNumericLiterals: ->
     @start = @position
     if /\d/.test @line[@position]
-      numberGroups = /(\d+)(\.\d+)?/.exec @line[@position..]
-      # if we grouped anything in the second group
-      # (numbers after the decimal point), then it is a floatlit
-      # if we didn't, it is a intlit
-      kind = if numberGroups[2] then 'FLOATLIT' else 'INTLIT'
-      @addToken {kind, lexeme: numberGroups[0]}
-      @position += numberGroups[0].length
+      numberGroups = /(?:\d+)(?:\.\d+)?(?:E[+-]?(?:\d+)(?:\.\d+)?)?/.exec @line[@position..]
+      # A number is considered a float if either the fraction or exponent
+      # part of the number has a decimal point. If the exponent is raised
+      # to a negative power, the number is also considered a float, since
+      # raising the exponent to a negative number implies moving the
+      # decimal point to the left.
+      numLit = numberGroups[0]
+      isFloat = '.' in numLit or '-' in numLit
```

Comment:
@rachelriv This is a broad check to capture things like **23E-1**. This is **2.3**. What about the number **100E-1**? I interpret this to be the number **10.0**. Why? The *E-1* part took the implicit decimal point and shifted it to the left by one. However, I don't consider the number **420E2** to be a float because when we shift the decimal point to the right, we only gain place holder zeros. **420E2** will look like **42000.**.

I include the check for the `-` because our num literals aren't signed when we scan them, so the only place they can have the `-` is in the exponent.